### PR TITLE
[OSS] Fix for torch dist broadcast randomly failing on dummy object

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Hopefully fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/5646
The doc for this feature was not super clear, looks like even if the object is to be overwritten it should not be really empty.
No logic change, minor tweak, basically making sure that we don't rely on the broadcast_list() util to catch empty objects
 

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

cc @SeanNaren @robogast

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: fairscale/optim/oss.py [hunk 1]
```diff
@@ -327,6 +327,9 @@ def _broadcast_state_dict(self) -> None:
             self.local_state_dict(), non_blocking=True, device=torch.device("cpu")
         )
 
+        # Tensor cannot be really empty, even if its size is meaningless
+        sync_tensor = torch.tensor([1], device=self._device)
```
Comment (min-xu-ai) [inline fairscale/optim/oss.py:331]: rename to dummy_sync_tensor?

File: pyproject.toml [hunk 2]
```diff
@@ -28,4 +28,4 @@ use_parentheses = true
 skip_glob = ["build/*", "stubs/*"]
 # Don't split "import" and "from".
 force_sort_within_sections = true
-known_third_party = ["datasets", "golden_configs", "helpers", "models", "numpy", "pytest", "recommonmark", "setuptools", "torch", "torch_pg", "torchtext", "torchvision"]
+known_third_party = ["benchmark_dataset", "datasets", "golden_configs", "helpers", "models", "numpy", "pytest", "recommonmark", "setuptools", "torch", "torch_pg", "torchtext", "torchvision"]
```
Comment (min-xu-ai) [inline pyproject.toml:31]: I never understood how to update this line. If you have some ideas, I'd love to learn. Otherwise, never mind.

**Review by min-xu-ai — APPROVED** [review]
very nice

File: pyproject.toml (hunk 2 above)
Comment (blefaudeux) [reply pyproject.toml:31]: I'm not doing anything, I'm using pre-commit as suggested in the contributing guide and it's doing that automatically

File: fairscale/optim/oss.py [hunk 3]
```diff
@@ -327,6 +327,9 @@ def _broadcast_state_dict(self) -> None:
             self.local_state_dict(), non_blocking=True, device=torch.device("cpu")
         )
 
+        # Tensor cannot be really empty, even if its size is meaningless
+        dummy_sync_tensor = torch.tensor([1], device=self._device)
```
Comment (robogast) [inline fairscale/optim/oss.py:331]: Shouln't the device of the dummy tensor match with the device of the stateful list?
What happens in `broadcast_object_list` when these have different devices?

**Pull request merged**